test/test_assetstore_model_override.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+from girder.models.file import File
+from girder.models.model_base import Model
+from girder.utility import assetstore_utilities
+from girder.utility.model_importer import ModelImporter
+from girder.utility.abstract_assetstore_adapter import AbstractAssetstoreAdapter
+
+
+class Fake(Model):
+    def initialize(self):
+        self.name = 'fake_collection'
+
+    def validate(self, doc):
+        return doc
+
+
+class FakeAdapter(AbstractAssetstoreAdapter):
+    def __init__(self, assetstore):
+        self.the_assetstore = assetstore
+
+
+@pytest.fixture
+def fakeModel(db):
+    ModelImporter.registerModel('fake', Fake(), plugin='fake_plugin')
+
+    yield Fake().save({
+        'foo': 'bar',
+        'type': 'fake'
+    })
+
+    ModelImporter.unregisterModel('fake', plugin='fake_plugin')
+
+
+@pytest.fixture
+def fakeAdapter(db):
+    assetstore_utilities.setAssetstoreAdapter('fake', FakeAdapter)
+
+    yield
+
+    assetstore_utilities.removeAssetstoreAdapter('fake')
+
+
+def testAssetstoreModelOverride(fakeModel, fakeAdapter, admin):
+    file = File().createFile(
+        creator=admin, item=None, name='a.out', size=0, assetstore=fakeModel,
+        assetstoreModel='fake', assetstoreModelPlugin='fake_plugin')
```

Comment:
Can you help me understand why we have to pass in the model 'name' ( 'fake') as well as the model instance here? Couldn't  the name be derived from the information provided in the ```registerModel``` call?

Reply:
The variable name `fakeModel` is misleading, the fixture is for registering an actual model class, but what the fixture actually returns is a document of that model, which is just a dict. Due to this stringly-typed document representation in Girder, we have to pass around model types out-of-band.

Reply:
I could always change the fixture so that it doesn't create a Fake document, and do that in the test function, if that would be cleaner.

Reply:
Ok, thanks for the explanation. fakeModel => fakeAssetstore, might make things a little clearer.